project.nix

```diff
@@ -1,211 +1,120 @@
+    })
+    gitignoreSource;
+  sources = {
+    neuron = "${gitignoreSource ./.}/neuron";
+    rib = import ./dep/rib/thunk.nix;
+    commonmark = import ./dep/commonmark-hs/thunk.nix;
+    reflex-dom-pandoc = import ./dep/reflex-dom-pandoc/thunk.nix;
   };
-  inherit (import gitignoreSrc { inherit (pkgs) lib; }) gitignoreSource;
-
-  commonmark = hackGet ./dep/commonmark-hs;
-  test-framework = hackGet ./dep/test-framework;
-
-  neuronSrc = pkgs.lib.cleanSource (gitignoreSource ./neuron);
 
-  neuronSearchScript = pkgs.runCommand "neuron-search" { buildInputs = [ pkgs.makeWrapper ]; } 
-    ''
+  searchBuilder = ''
     mkdir -p $out/bin
-    makeWrapper ${neuronSrc}/src-bash/neuron-search $out/bin/neuron-search --prefix 'PATH' ':' \
-        "${pkgs.fzf}/bin:${pkgs.ripgrep}/bin:${pkgs.gawk}/bin:${pkgs.bat}/bin:${pkgs.findutils}/bin:${pkgs.envsubst}/bin"
-    '';
-
-in {
-  inherit withHoogle;
-
-  shellToolOverrides = ghc: super: {
-    inherit neuronSearchScript;
+    cp $src/src-bash/neuron-search $out/bin/neuron-search
+    chmod +x $out/bin/neuron-search
+    wrapProgram $out/bin/neuron-search --prefix 'PATH' ':' ${
+      with pkgs;
+      lib.makeBinPath [ fzf ripgrep gawk bat findutils envsubst ]
+    }
+    PATH=$PATH:$out/bin
+  '';
+  wrapSearchScript = drv: {
+    buildTools = [ pkgs.makeWrapper ];
+    preConfigure = searchBuilder;
   };
 
-  packages = {
-    neuron = neuronSrc;
-    # TODO: expose these overrides so it can be used in the other project
-    #that uses neuron as a thunk.
-
-    reflex-dom-pandoc = hackGet ./dep/reflex-dom-pandoc;
-    shake = hackGet ./dep/shake;
-    dependent-sum-aeson-orphans = hackGet ./dep/dependent-sum-aeson-orphans;
-    
-    # commonmark
-    commonmark = commonmark + "/commonmark";
-    commonmark-extensions = commonmark + "/commonmark-extensions";
-    commonmark-pandoc = commonmark + "/commonmark-pandoc";
-    emojis = hackGet ./dep/emojis;
-    pandoc-types = hackGet ./dep/pandoc-types;
-    texmath = hackGet ./dep/texmath;
-    hslua = hackGet ./dep/hslua;
-    doctemplates = hackGet ./dep/doctemplates;
-    doclayout = hackGet ./dep/doclayout;
-    jira-wiki-markup = hackGet ./dep/jira-wiki-markup;
-
-    test-framework = test-framework + "/core";
-    # test-framework-th = null;
-    test-framework-hunit = test-framework + "/hunit";
-    test-framework-quickcheck2 = test-framework + "/quickcheck2";
-
-    # pandoc
-    pandoc = hackGet ./dep/pandoc;
-    regex-base = hackGet ./dep/regex-base;
-    regex-posix = hackGet ./dep/regex-posix;
-    regex-pcre = hackGet ./dep/regex-pcre;
-    regex-pcre-builtin = hackGet ./dep/regex-pcre-builtin;
-
-    algebraic-graphs = hackGet ./dep/alga;
-    clay = hackGet ./dep/clay;
-    HsYAML = hackGet ./dep/HsYAML;
-
-    rib = hackGet ./dep/rib;
+  overrides = self: super: {
+    # We include rib because it is quite tightly coupled with neuron development
+    rib = doJailbreak (self.callCabal2nix "rib" sources.rib { });
```

Comment:
Why do we need to jailbreak `rib`?

Reply:
rib pins `dhall 1.30.*` and `relude 0.6.*` apparently it builds with the newer nixpkgs. So I guess we should loosen the constraints in rib and bump the pin?

Reply:
Yup, definitely favour loosening the constraints on rib and reflex-dom-pandoc.

Reply:
@maralorn This was already done? https://github.com/srid/rib/commit/42bc7c3e6b092cb1640b078b772ac09720cf8825

You can update the rib thunk using obelisk.

Reply:
Yeah, cool.

Reply:
What do you think about tracking nixpkgs via an ob thunk as well?